### PR TITLE
ci: unversion workflows

### PR DESCRIPTION
**Pull request by dannysauer**
## Description

Instead of hard-coding version strings in workflows, read the VERSION file and set an env var for that.

## Test Plan

Run the workflows?

## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.
- [x] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

### CODE REVIEW

**Comment by dannysauer** [comment]
Dang it.  Git is hard. :p

File: .github/workflows/publish-helm.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            printf 'CHART_URL=%s' \
+              "$( gh api "$ASSET_URL" -q '.browser_download_url')" >> $GITHUB_ENV
+
+        - name: update index
+          # helm repo index . --merge index.yaml
+        - name: replace chart file with redirect
+          # https://stackoverflow.com/questions/39657226/jekyll-redirect-link
+        - name: commit back
+
+# if release
```
Comment (loksonarius) [inline .github/workflows/publish-helm.yml:53]: Leftovers?

File: .github/workflows/publish-helm.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+---
+name: publish-helm-chart
+
+on:  # yamllint disable-line rule:truthy
+  workflow-dispatch:
+    inputs:
+      tag:
+        description: "Release Tag"
+        required: "true"
+        default: "0.0.0"
```
Comment (loksonarius) [inline .github/workflows/publish-helm.yml:10]: Should this instead match the current release version? If this is a hard coded string, it'll be bumped during automatic version bumps.

**Review by loksonarius — CHANGES_REQUESTED** [review]

File: .github/workflows/publish-helm.yml (hunk 1 above)
Comment (dannysauer) [reply .github/workflows/publish-helm.yml]: These commits belong to a different branch. :rofl: 
https://github.com/determined-ai/determined/pull/4753

**Comment by loksonarius** [comment]
I'm not sure this PR captures the intended changes. My review's for the new workflow, but the PR title would make me figure there should be updates across multiple workflows

**Review by loksonarius — APPROVED** [review]

**Comment by dannysauer** [comment]
I always forget to fix the title on these. :D

Also, apparently I forgot how to use quotes in the shell.  Forcepush again fixes that.

**Comment by dannysauer** [comment]
Ok, one more time, now with more Windows-friendly shell commands.  Grumble.

**Comment by dannysauer** [comment]
Ninth time's the charm? :/

**Comment by dannysauer** [comment]
Ok, finally I learned How To Windows and everything works.  Except the required CircleCI checks, but that's because CircleCI is currently having technical problems.  Those are unrelated to this change anyway, so I'm force-merging. :)

**Pull request merged**